### PR TITLE
Uevent-based matching for unix-char and unix-block

### PR DESCRIPTION
**Pull request by brauner**
This is a rebased version of https://github.com/lxc/lxd/pull/6530 .

### CODE REVIEW

**Comment by brauner** [comment]
@Lilyjjo, @splett2 fyi

**Comment by stgraber** [comment]
```
-				ueventBuf = ueventBuf[40:len(ueventBuf)-1]
+				ueventBuf = ueventBuf[40 : len(ueventBuf)-1]
```

@brauner 

(I know it's my fault :))

**Comment by brauner** [comment]
> ```
> -				ueventBuf = ueventBuf[40:len(ueventBuf)-1]
> +				ueventBuf = ueventBuf[40 : len(ueventBuf)-1]
> ```
> 
> @brauner
> 
> (I know it's my fault :))

No, I didn't call `go fmt`. :)

**Comment by brauner** [comment]
> > ```
> > -				ueventBuf = ueventBuf[40:len(ueventBuf)-1]
> > +				ueventBuf = ueventBuf[40 : len(ueventBuf)-1]
> > ```
> > 
> > 
> > @brauner
> > (I know it's my fault :))
> 
> No, I didn't call `go fmt`. :)

fixed

File: doc/api-extensions.md [hunk 1]
```diff
@@ -902,3 +902,7 @@ This adds the ability to use LVM stripes on normal volumes and thin pool volumes
 
 ## vm\_boot\_priority
 Adds a `boot.priority` property on nic and disk devices to control the boot order.
+Allows a list of profiles to be applied to an image when launching a new container.
```
Comment (stgraber) [inline doc/api-extensions.md]: Bad merge ^

File: doc/instances.md [hunk 2]
```diff
@@ -720,6 +721,22 @@ security.gid    | int       | 0             | no        | What GID to drop privi
 lxc config device add <instance> <device-name> proxy listen=<type>:<addr>:<port>[-<port>][,<port>] connect=<type>:<addr>:<port> bind=<host/instance>
 ```
 
+### Type: unix-hotplug
```
Comment (stgraber) [inline doc/instances.md]: Needs a line saying it's container only (see other types for example).

**Pull request merged**